components/list/list-item-checkbox-mixin.js

```diff
@@ -0,0 +1,123 @@
+import { css, html } from 'lit-element/lit-element.js';
+import { checkboxStyles } from '../inputs/input-checkbox-styles.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+
+export const ListItemCheckboxMixin = superclass => class extends superclass {
+
+	/**
+	 * https://lit-element.polymer-project.org/guide/properties#declare
+	 *
+	 * @readonly
+	 * @static
+	 */
+	static get properties() {
+		return {
+			disabled: {type: Boolean },
+			key: { type: String, reflect: true },
+			selectable: {type: Boolean },
+			selected: { type: Boolean, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return [ checkboxStyles, css`
+		.list-item-checkbox {
+			grid-column: control-start / control-end;
+			grid-row: 1 / 2;
+		}
+		.list-item-checkbox-action {
+			grid-column: control-start / end;
+			grid-row: 1 / 2;
+			z-index: 2;
+			cursor: pointer;
+		}
+		:host([disabled]) .list-item-checkbox-action {
+			cursor: default;
+		}
+		`];
+	}
+
+	constructor() {
+		super();
+		this._checkboxId = getUniqueId();
+	}
+
+	/**
+	 * Dispatches a custom event with information on what
+	 * items are selected and their keys.
+	 * https://lit-element.polymer-project.org/guide/events
+	 *
+	 * @param {*} value New state of checkbox
+	 */
+	_dispatchSelected(value) {
```

Comment:
Method order. 

https://github.com/Brightspace/eslint-config-brightspace/blob/master/sort-member-config.js

Not sure why the linter isn't picking this up.

Reply:
Yeah I don't think that plugin has been working... at one point it was.

Reply:
I pretty much never remember to do this when it isn't caught by the linter.

Reply:
same but it makes the coding experience so much better. I now have muscle memory of where parts of the code should be and get anxiety when it isn't.